src/Net/Http.fs

```diff
@@ -554,20 +554,39 @@ module private HttpHelpers =
         hasContentType.Value
 
     let getResponse (req:HttpWebRequest) silentHttpErrors =
-        if defaultArg silentHttpErrors false then
+
+        let getResponseAsync (req:HttpWebRequest) =
             async {
-                try
+                #if FX_NO_WEBREQUEST_TIMEOUT
                     return! Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
-                with
-                    | :? WebException as exn -> 
-                        if exn.Response <> null then 
-                           return exn.Response
-                        else 
-                            reraisePreserveStackTrace exn
-                            return Unchecked.defaultof<_>
+                #else
+                    let! child = 
+                        Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
+                        |> fun x -> Async.StartChild(x, req.Timeout)
```

Comment:
I see. I do like the idea of having optional `timeout` parameter - that sounds like something that would be quite useful for many people anyway!

That said, if the value is the default, then it probably should be `Timeout.Infinite`, so we could just check for that and use async only when it is not infinite - that should work too.

I'm happy with either of the options.